### PR TITLE
Task1- Add shipment_items count by description function

### PR DESCRIPTION
**Pull request by cdragon1116**
Main Task -
Add shipment_items count by description function
Add corresponding rspec

[Ticket: Task1]

### CODE REVIEW

**Comment by cdragon1116** [comment]
@tomohung , please help check Task1 PR, many thanks!

**Comment by alChaCC** [comment]
👍

File: app/models/shipment.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Shipment < ApplicationRecord
   belongs_to :company
   has_many :shipment_items
+
+  def items_count_by_description
+    items = shipment_items.group('description').order('count(*) desc').count
```
Comment (tomohung) [inline app/models/shipment.rb]: 👍 impressive

File: spec/models/shipment_spec.rb [hunk 2]
```diff
@@ -1,5 +1,24 @@
 require "rails_helper"
 
 RSpec.describe Shipment, type: :model do
-  # write your tests here
+  let(:shipment) { create(:shipment) }
+
+  describe "items_count_by_description function test" do 
+    it "should be count by description and ordered by count in desc. order" do 
+      2.times{ shipment.shipment_items.create!( description: 'iPhone') }
```
Comment (tomohung) [inline spec/models/shipment_spec.rb]: You can check FactoryBot `create_list` for these data

File: Gemfile [hunk 3]
```diff
@@ -15,6 +15,8 @@ group :development, :test do
   gem 'byebug', platform: :mri
   gem 'faker'
   gem 'pry'
+  gem 'factory_bot_rails', '~> 5.0', '>= 5.0.2'
```
Comment (tomohung) [inline Gemfile]: you should have other commits for these changes

File: spec/models/shipment_spec.rb (hunk 2 above)
Comment (cdragon1116) [reply spec/models/shipment_spec.rb]: Thanks for suggestions, I'll study the usage of `create_list` and  change it on next commit.

File: Gemfile (hunk 3 above)
Comment (cdragon1116) [reply Gemfile]: Got it ! I'll put the whole spec test to another commit next time. 
Thanks for reminding!

**Pull request merged**